src/test/java/com/hivemq/persistence/local/memory/ClientSessionMemoryLocalPersistenceTest.java

```diff
@@ -0,0 +1,764 @@
+
+    private ClientSessionLocalPersistence persistence;
+
+    @Mock
+    private LocalPersistenceFileUtil localPersistenceFileUtil;
+
+    @Mock
+    private MqttConfigurationService mqttConfigurationService;
+
+    @Mock
+    private PublishPayloadPersistence payloadPersistence;
+
+    @Mock
+    private EventLog eventLog;
+    private MetricRegistry metricRegistry;
+    private Gauge memoryGauge;
+
+    @Before
+    public void before() throws Exception {
+        MockitoAnnotations.initMocks(this);
+
+        InternalConfigurations.PERSISTENCE_CLOSE_RETRIES.set(3);
+        InternalConfigurations.PERSISTENCE_CLOSE_RETRY_INTERVAL.set(5);
+        InternalConfigurations.PERSISTENCE_BUCKET_COUNT.set(BUCKET_COUNT);
+        when(localPersistenceFileUtil.getVersionedLocalPersistenceFolder(anyString(), anyString())).thenReturn(
+                temporaryFolder.newFolder());
+
+        metricRegistry = new MetricRegistry();
+        persistence = new ClientSessionMemoryLocalPersistence(mqttConfigurationService, payloadPersistence, metricRegistry, eventLog);
+        memoryGauge = metricRegistry.gauge(HiveMQMetrics.CLIENT_SESSIONS_MEMORY_PERSISTENCE_TOTAL_SIZE.name(), null);
+    }
+
+    @Test
+    public void test_put_get() {
+
+        persistence.put("clientid",
+                new ClientSession(false, SESSION_EXPIRY_MAX),
+                123L,
+                BucketUtils.getBucket("clientid", BUCKET_COUNT));
+
+        final ClientSession clientSession =
+                persistence.getSession("clientid", BucketUtils.getBucket("clientid", BUCKET_COUNT));
+
+        assertEquals(false, clientSession.isConnected());
+
+        final ClientSession session = persistence.getSession("clientid");
+        assertNotNull(session);
+
+        assertEquals(123L, persistence.getTimestamp("clientid").longValue());
+        assertTrue((long) memoryGauge.getValue() > 0);
+    }
+
+    @Test
+    public void test_getDisconnected() {
+
+        final String client1 = TestBucketUtil.getId(1, BUCKET_COUNT);
+        final String client2 = TestBucketUtil.getId(1, BUCKET_COUNT);
+
+        persistence.put(client1, new ClientSession(true, SESSION_EXPIRY_MAX), 123L, 1);
+        persistence.put(client2, new ClientSession(true, SESSION_EXPIRY_MAX), 123L, 1);
+
+        persistence.disconnect(client2, 124L, false, 1, SESSION_EXPIRY_MAX);
+
+        final long memory = (long) memoryGauge.getValue();
+        assertTrue(memory > 0);
+
+        final Set<String> disconnectedClients = persistence.getDisconnectedClients(1);
+
+        assertEquals(1, disconnectedClients.size());
+        assertTrue(disconnectedClients.contains(client2));
+        assertEquals(memory, (long) memoryGauge.getValue());
+    }
+
+    @Test
+    public void test_getDisconnectedClients() {
+
+        final String client1 = TestBucketUtil.getId(1, BUCKET_COUNT);
+
+        persistence.put(client1, new ClientSession(false, SESSION_EXPIRY_MAX), 123L, 1);
```

Comment:
if you want a disconnected session in a test, than put a connected session and disconnect it afterwards. Counts for all tests